vizro-core/tests/unit/vizro/models/_components/test_container.py

```diff
@@ -217,3 +219,24 @@ def test_container_build_with_description(self):
                 id="container_title_content",
             ),
         )
+
+    def test_container_build_with_controls(self):
+        result = vm.Container(
+            id="container", components=[vm.Button()], controls=[vm.Filter(column="species", selector=vm.Dropdown())]
+        ).build()
+        assert_component_equal(
+            result, dbc.Container(id="container", class_name="", fluid=True), keys_to_strip={"children"}
+        )
+        assert_component_equal(result.children, [None, html.Div(), html.Div()], keys_to_strip=STRIP_ALL)
+        # Test the exact Div produced in Container.build:
+        assert_component_equal(
+            result.children[1],
+            html.Div(
+                id="container-control-panel",
+                children=[],
+                className="container-controls-panel",
+            ),
+            keys_to_strip={"children"},
+        )
+        # Test if correct selector is added
+        assert isinstance(result.children[1].children[0].children[1], dcc.Dropdown)
```

Comment:
These tests are nicer if we can minimise reference to `children[1]` etc. and instead use `id` to refer to components. See the changes I made here for an example: https://github.com/mckinsey/vizro/pull/1169/commits/f10e91bd5152b7f67773fa96af7ffca4c0d3c661

So this would become something like this:
```suggestion
        assert_component_equal(
            result["container-control-panel"],
            html.Div(
                id="container-control-panel",
                className="container-controls-panel",
            ),
            keys_to_strip={"children"},
        )
        assert_component_equal(result["dropdown-id"], dcc.Dropdown(), keys_to_strip=STRIP_ALL)
```

where you'll need to specify `vm.Dropdown(id="dropdown_id")`